### PR TITLE
feat: Added image paste feature (resolves #856)

### LINKED ISSUE #856: Add image paste
**Is your feature request related to a problem? Please describe.**
The modern chats contain the functionality of pasting not only text, but the images. Unavailability of this function leads to additional steps to be undertaken by potential user and thus make the app less comfortable for chatting.

**Is your feature request related to a specific XEP?**
XEP-0363

**Describe the solution you'd like**
Add pasting the image directly in the window of a chat. When pasted it is automatically http-uploaded to the server and ready to be sent to an interlocutor or muc.

**Additional context**
Furthermore additional tools on resizing, turning image may be added.

### PR DESCRIPTION
**Pull request by git-it**
JSXC is great! I'd like to help JSXC stay awesome :) 

This merge hopefully resolves #856 via a paste event trigger from the chatWindow input. This event triggers a check if the clipboard contains an image. If so the clipboard is read in as a blob and processed using the existing methods for file uploading.

### CODE REVIEW

File: src/ui/ChatWindowFileTransferHandler.ts [hunk 1]
```diff
@@ -3,12 +3,15 @@ import Attachment from '../Attachment'
...
 export default class FileTransferHandler {
    private handlerElement;
+   private inputElement;
 
    constructor(private chatWindow: ChatWindow) {
       this.handlerElement = this.chatWindow.getDom().find('.jsxc-file-transfer');
 
       this.handlerElement.on('click', this.showFileSelection);
 
+      this.inputElement.on('paste', this.pasteImageAsFile);
```
Comment (sualko) [inline src/ui/ChatWindowFileTransferHandler.ts:13]: Where gets the inputElement defined? At this point it should be undefined.

**Review by sualko — CHANGES_REQUESTED** [review]
Thanks for your contribution and the kind words. It's nice to see that someone is enjoying jsxc.

File: src/ui/ChatWindowFileTransferHandler.ts (hunk 1 above)
Comment (git-it) [reply src/ui/ChatWindowFileTransferHandler.ts:14]: you are absolutely correct. 
Fixed this with commit 19c37a4719e688ed6a17fb96b98da3ff00444543. Line 10 gets the element from the dom (using the same method as done in chatWindow).
Thanks for your patience and guidance

**Comment by sualko** [comment]
Thanks again for this nice feature :tada: :+1: Great work!

**Pull request merged**